File: Data Structures/Linked Lists/Delete a Node.cpp
/*
   Contributer : github.com/Blind Bluffer
   Email : shivank309@gmail.com
*/
/*
  Delete Node at a given position in a linked list 
  Node is defined as 
  struct Node
  {
     int data;
     struct Node *next;
  }
*/
Node* Delete(Node *head, int position)
{
  // Complete this method
    Node* temp;
    Node* t;
    if(position==0){
        t=head;
        head=t->next;
        delete t;
    }
    else{
        int i=0;
        temp=head;
        while(i!=(position-1)){
            temp=temp->next;
            i++;
        }
        t=temp->next;
        temp->next=t->next;
        delete t;
    }
    
    return head;
}

File: Data Structures/Linked Lists/Delete a Node.java
/* Contributer : github.com/Shaurya9619
	   Email : shaurya96gairola@gmail.com
	*/
Node Delete(Node head, int position) {
    

    Node temp;
    Node t;
    if(position==0){
        t=head;
        head=t.next;
        t=null;
        System.gc();
    }
    else{
        int i=0;
        temp=head;
        while(i!=(position-1)){
            temp=temp.next;
            i++;
        }
        t=temp.next;
        temp.next=t.next;
       t=null;
        System.gc();
    }
    
    return head;
}
